read_spec_dr7.m

```diff
@@ -0,0 +1,43 @@
+% in the same units as the spectrum),
+% the forth row is the mask array.
+%
+% The spectra are binned log-linear. Units are 10^(-17) erg/cm^2/s/Ang
+% http://classic.sdss.org/dr7/products/spectra/read_spSpec.html
+% Notice that the wavelength vector is not contained in the image,
+% but must be generated from parameters in the header. SDSS spectra
+% are binned in constant Log(Λ) and the wavelength can be obtained
+% from the header parameters COEFF0 and COEFF1 (or alternatively
+% CRVAL1 and CD1_1) as follows:
+%
+% Λ = 10**(COEFF0 + COEFF1*i), where i denotes the (zero indexed) pixel number.
+%
+% Remember that these are vacuum wavelengths.
+%
+
+function [wavelengths, flux, noise_variance, pixel_mask] = read_spec_dr7(filename)
+
+measurements = fitsread(filename);
+%           'binarytable',  1, ...
+%           'tablecolumns', 1:4);
+
+% the Primary table contains spectra
+
+% acquire un-continuum subtracted spectrum
+flux = measurements(1,:);
+
+% noise (standard deviation)
+noise = measurements(3,:);
+noise_ratio = noise./flux;
+noise_variance = noise.^2;
+% mask array
+and_mask = measurements(4,:);
+
+% acquire rest wavelength
+header = fitsinfo(filename);
+coef0 = header.PrimaryData.Keywords{19,2}; % Center wavelength (log10) of first pi
+coef1 = header.PrimaryData.Keywords{20,2}; % Log10 dispersion per pixel
+
+length = numel(flux);
+
+wavelengths = 10.^(linspace(coef0, coef0 + coef1*(length + 1), length));
+pixel_mask =  (noise_ratio>=2) | (and_mask==0x800000);
```

Comment:
`(and_mask==0x800000)` is not a valid expression in MATLAB.

Reply:
Later than 2019b, it works https://www.mathworks.com/help/matlab/matlab_prog/specify-hexadecimal-and-binary-numbers.html